eng/pipelines/templates/BuildAndTest.yml

```diff
@@ -21,6 +21,7 @@ parameters:
 steps:
   - script: ${{ parameters.buildScript }}
             -restore -build
+            -pack
```

Comment:
Good point. This pack could conflict with the subsequent pack, since that one is signed. I will investigate this in a separate PR. And it will really be needed only when we have this running on windows again.
For now, I will add `-pack` only for linux, where the separate `pack` step is not defined at all.